src/com/SocketSendQueue.cpp

```diff
@@ -0,0 +1,35 @@
+#include "SocketSendQueue.hpp"
+
+namespace precice
+{
+namespace com
+{
+namespace asio = boost::asio;
+
+void SendQueue::push(std::shared_ptr<Socket> sock, 
```

Comment:
This function is called `push` but does `push_and_process`. I think the caller should decide whether to process the queue or not by explicitly calling `process()`.

As this function modifies the underlying queue, so it should acquire the `_queueMutex` first.